src/Sarif/Core/Result.cs

```diff
@@ -17,9 +17,15 @@ public partial class Result
         /// </remarks>
         public Run Run { get; set; }
 
-        public bool ShouldSerializeWorkItemUris() { return this.WorkItemUris != null && this.WorkItemUris.Any((s) => s != null); }
+        public bool ShouldSerializeWorkItemUris()
+        {
+            return this.WorkItemUris != null && this.WorkItemUris.Any((s) => s != null);
+        }
 
-        public bool ShouldSerializeLevel() { return this.Level != FailureLevel.Warning; }
+        public bool ShouldSerializeLevel()
+        {
+            return this.Level != FailureLevel.Warning;
```

Comment:
>FailureLevel.Warning [](start = 33, length = 20)

This is wrong, we should only set this if level doesn't correspond to the default rule level. Also, this value relates to kind! So, if this isn't a failure result, level == 0.